alarm_trap_sender.cpp

```diff
@@ -50,22 +50,36 @@ AlarmFilter AlarmFilter::_instance;
 
 AlarmTrapSender AlarmTrapSender::_instance;
 
-bool ObservedAlarms::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
+bool ObservedAlarms::update(const AlarmTableDef& alarm_table_def, const std::string& issuer)
 {
-  bool updated = false;
+  if (!is_active(alarm_table_def))
+  {
+    _index_to_entry[alarm_table_def.alarm_index()] = AlarmListEntry(alarm_table_def, issuer);
+    return true;
```

Comment:
I quite liked the `updated = true; ... return updated` because it was obvious what the return value meant.